### PR TITLE
(sample): Device streaming sample improvements and fixes

### PR DESCRIPTION
**Pull request by vinagesh**
Updated the sample with some improvements

### CODE REVIEW

File: iot-hub/Samples/device/DeviceStreamingSample/DeviceClientStreamingSample.csproj [hunk 1]
```diff
@@ -16,6 +16,6 @@
   </ItemGroup>
   
   <ItemGroup>
-	<PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.29.0-preview-*" />
+	<PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.32.0-preview-001" />
```
Comment (vinagesh) [inline iot-hub/Samples/device/DeviceStreamingSample/DeviceClientStreamingSample.csproj:19]: Upgraded sample to use the latest version where this sample will work for net 5.0 as well

File: iot-hub/Samples/service/DeviceStreamingSample/DeviceStreamSample.cs [hunk 2]
```diff
@@ -44,6 +44,8 @@ public async Task RunSampleAsync()
 
                     var receiveResult = await stream.ReceiveAsync(receiveBuffer, cts.Token).ConfigureAwait(false);
                     Console.WriteLine($"Received stream data: {Encoding.UTF8.GetString(receiveBuffer, 0, receiveResult.Count)}");
+
+                    await stream.CloseAsync(WebSocketCloseStatus.NormalClosure, "Streaming completed", new CancellationToken());
```
Comment (vinagesh) [inline iot-hub/Samples/service/DeviceStreamingSample/DeviceStreamSample.cs:48]: The sample throws and exception without this

File: iot-hub/Samples/service/DeviceStreamingSample/Program.cs [hunk 3]
```diff
@@ -1,53 +1,31 @@
...
-                string.IsNullOrEmpty(s_deviceId))
-            {
-                Console.WriteLine("Please provide a connection string and device ID");
-                Console.WriteLine("Usage: ServiceClientC2DStreamingSample [iotHubConnString] [deviceId]");
-                return 1;
-            }
-
-            using ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(s_connectionString, s_transportType);
-            var sample = new DeviceStreamSample(serviceClient, s_deviceId);
+            // Parse application parameters
```
Comment (vinagesh) [inline iot-hub/Samples/service/DeviceStreamingSample/Program.cs:15]: Adding command line arguments for inputs

File: iot-hub/Samples/service/DeviceStreamingSample/Parameters.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            'd',
+            "DeviceId",
+            Required = true,
+            HelpText = "The Id of the device to connect to.")]
+        public string DeviceId { get; set; }
+
+        [Option(
+            't',
+            "TransportType",
+            Default = TransportType.Amqp,
```
Comment (drwill-ms) [inline iot-hub/Samples/service/DeviceStreamingSample/Parameters.cs:31]: Can we default to MQTT?

File: iot-hub/Samples/service/DeviceStreamingSample/Program.cs [hunk 5]
```diff
@@ -1,53 +1,31 @@
...
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            using ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(parameters.HubConnectionString, parameters.TransportType);
+            var sample = new DeviceStreamSample(serviceClient, parameters.DeviceId);
             await sample.RunSampleAsync().ConfigureAwait(false);
```
Comment (drwill-ms) [inline iot-hub/Samples/service/DeviceStreamingSample/Program.cs:29]: Let's remove `.ConfigureAwait(false)` calls from samples.

**Review by drwill-ms — APPROVED** [review]

File: iot-hub/Samples/service/DeviceStreamingSample/Parameters.cs (hunk 4 above)
Comment (vinagesh) [reply iot-hub/Samples/service/DeviceStreamingSample/Parameters.cs:31]: The TransportType enum in our service client does not have MQTT.  Are the different protocols only for the device client?

**Review by azabbasi — APPROVED** [review]

File: iot-hub/Samples/service/DeviceStreamingSample/Parameters.cs (hunk 4 above)
Comment (drwill-ms) [reply iot-hub/Samples/service/DeviceStreamingSample/Parameters.cs:28]: Oops, good point.

**Pull request merged**